mango-core/src/main/java/org/calrissian/mango/net/MoreInetAddresses.java

```diff
@@ -321,6 +323,41 @@ public static Inet6Address decrement(Inet6Address ip) {
         return getInet6Address(decrementBytes(ip));
     }
 
+    /**
+     * Returns {@code true} if {@code address} is within the bounds of the {@code cidr}.
+     *
+     * @param address   the {@link InetAddress} to check
+     * @param cidr      the {@link CidrInfo} block to check bounds against
+     * @return          {@code true} if {@code address} is within the {@code cidr} range. {@code false} otherwise.
+     */
+    public static Boolean isInCidrRange(InetAddress address, CidrInfo cidr) {
+
+        if (!(address.getClass().equals(cidr.getNetwork().getClass()))) {
+            throw new IllegalArgumentException(
+                    format("CIDR [%s] is of different INET Address family compared to address [%s]",
+                            cidr,
+                            toAddrString(address)));
+        }
+
+        Range<BigInteger> cidrRange =
```

Comment:
In the previous check you validate that the cidr network address is of the same type, then why not simply convert to IP types and do a contains check?

Otherwise if you want to deal with raw types then simply use byte array comparitor instead of converting to BigInteger.